include/assimp/MemoryIOWrapper.h

```diff
@@ -162,7 +157,11 @@ class MemoryIOSystem : public IOSystem {
     }
 
     /** Destructor. */
-    ~MemoryIOSystem() = default;
+    ~MemoryIOSystem() {
+        for (auto &it : created_streams) {
+            MemoryIOSystem::Close(it);
```

Comment:
`MemoryIOSystem::Close` modifies `created_streams` while the destructor is iterating over it.